frame/contracts/src/lib.rs

```diff
@@ -967,6 +989,156 @@ struct InternalOutput<T: Config, O> {
 	result: Result<O, ExecError>,
 }
 
+/// Helper trait to wrap contract execution entry points into a signle function
+/// [`Invokable::run_guarded`].
+trait Invokable<T: Config, O> {
```

Comment:
I think `O` should be an associated type instead of a type parameter: It is chosen by the implementation rather than by the caller. `Invokeable` is also only implemented for a single `O` per type.